src/connector/src/sink/elasticsearch_opensearch/elasticsearch_opensearch_formatter.rs

```diff
@@ -152,7 +154,30 @@ impl ElasticSearchOpenSearchFormatter {
             match op {
                 Op::Insert | Op::UpdateInsert => {
                     let key = self.key_encoder.encode(rows)?.into_string()?;
-                    let value = self.value_encoder.encode(rows)?;
+                    let mut modified_col_indices = Vec::new();
+                    if let Some((delete_key, delete_row)) = update_delete_row.take()
+                        && delete_key == key
```

Comment:
We may separate the handling of `Op::Insert` and `Op::UpdateInsert`, because we should only check the `update_delete_row` on `Op::UpdateInsert`, and it's expected to have  `Some(update_delete_row)`, and we should add warning log on `update_delete_row.is_none()` or `delete_key != key`, and on `Op::Insert` we can reuse the original logic and don't check the `update_delete_row` and don't do partial update.